pkg/providers/tinkerbell/template.go

```diff
@@ -78,6 +87,13 @@ func (tb *TemplateBuilder) GenerateCAPISpecWorkers(clusterSpec *cluster.Spec, wo
 	workerSpecs := make([][]byte, 0, len(clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations))
 	for _, workerNodeGroupConfiguration := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
 		wTemplateConfig := clusterSpec.TinkerbellTemplateConfigs[tb.WorkerNodeGroupMachineSpecs[workerNodeGroupConfiguration.MachineGroupRef.Name].TemplateRef.Name]
```

Comment:
What happens if I don't specify a template reference in the YAML? Won't this panic?

Reply:
templateRef field is an optional field so it won't panic.

Reply:
Sorry, I thought `TemplateRef` was a pointer but its not, its a literal. All good!